BAZProjectC1/CachedImageView.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  CachedImageView.swift
+//  BAZProjectC1
+//
+//  Created by 961184 on 19/09/22.
+//
+
+import Foundation
+import UIKit
+
+let imageCache = NSCache<NSString, UIImage>()
+class CachedImageView : UIImageView {
+    var imageUrl: String?
```

Comment:
Sugerencia. Y completamente opcional para esta entrega.
Remove el uso de subclases y associar esta propiedad a la instancia del UIImageView. 
https://medium.com/@marcosantadev/stored-properties-in-swift-extensions-615d4c5a9a58

```swift
extension UIImageView {
    private static let imageCache = NSCache<NSString, UIImage>()

    private struct AssociatedKeys {
        static var imageUrl: UInt8 = 0
    }

    var imageUrl: String? {
        get {
            objc_getAssociatedObject(self, &AssociatedKeys.imageUrl) as? String
        }
        set {
            objc_setAssociatedObject(self, &AssociatedKeys.imageUrl, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
        }

    }
}
```